### PR TITLE
[UTOPIA-567] Added lib Auth and custom Roles Guard affecting all endpoints

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
cc: [This PR will need OpenShift secret `KEYCLOAK_AUTH_SERVER_URI` changed to `<domain>/auth`]

# Description

This PR includes the following proposed change(s):

- AuthGuard added [nest-keycloak-connect]
- Custom RolesGuard created
- Removed access token check from controllers - which AuthGuard will take care of
- Removed AuthService from AuthModule exports and Other modules' imports
- Custom Request Interface - including user, userRoles, accessToken
- Fixed & updated tests after Auth and Role guard changes

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Tested via Swagger: 
Validated by adding roles to @Roles decorator and SSO portal;

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [x] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/common/interfaces/response.interface.ts
Comment (kushal-arora-fw) [reply src/backend/src/common/interfaces/response.interface.ts:3]: Initially it was `export interface IResponse extends Response = {}`

but, our Linting strategy does not allow empty Interface. And this came out as a recommended suggestion.

The other alternated was to disable Linting for the line, which I do not want to encourage. Hence. went ahead with the former approach.

Comment (Bottle7) [reply src/backend/src/common/interfaces/response.interface.ts:3]: Okay, sounds good 👍🏼 I 100% agree with not selectively disabling linting as a practice.

**Comment by Bottle7** [comment]
Should this have a 'DO NOT MERGE' tag until the change has been made to the OpenShift secret?

**Comment by kushal-arora-fw** [comment]
> Should this have a 'DO NOT MERGE' tag until the change has been made to the OpenShift secret?

Yeah, I can add. However, I would still need approval before I go ahead and change OS side of things - as both OS secrets change and this needs to go almost at same time. Going early will break existing dev, later will break new xD

**Review by Bottle7 — APPROVED** [review]

**Review by binggaofw — APPROVED** [review]

**Comment by kushal-arora-fw** [comment]
Dev Secret `KEYCLOAK_AUTH_SERVER_URI` updated to `https://dev.loginproxy.gov.bc.ca/auth`. Merging in.

**Pull request merged**
